### PR TITLE
Added new controllers

### PR DESCRIPTION
**Pull request by MykolaFatkullin**
- Created controllers: AuthenticationController, OrderController, ShoppingCartController, UserController;
- Created all necessary DTOs;
- Created mappers for all DTOs;
- Refactored GenericDto;

### CODE REVIEW

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/com/tickets/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    private final UserService userService;
+    private final UserMapper userMapper;
+
+    public UserController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getById(@RequestParam String email) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/controller/UserController.java]:
```suggestion
    public UserResponseDto getByEmail(@RequestParam String email) {
```

File: src/main/java/com/tickets/model/dto/OrderRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.tickets.model.dto;
+
+import java.util.List;
+
+public class OrderRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/model/dto/OrderRequestDto.java]: do we use this?

File: src/main/java/com/tickets/model/dto/OrderResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.tickets.model.dto;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/model/dto/OrderResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

File: src/main/java/com/tickets/model/dto/ShoppingCartResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.tickets.model.dto;
+
+import java.util.List;
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/model/dto/ShoppingCartResponseDto.java]: same

File: src/main/java/com/tickets/service/mapper/GenericDtoToMap.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package com.tickets.service.mapper;
+
+public interface GenericDtoToMap<T, V> {
+    V dtoToMap(T requestDto);
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/GenericDtoToMap.java]:
```suggestion
public interface GenericMapToEntity<T, V> {
    V mapToEntity(T requestDto);
```

File: src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OrderMapperImpl implements OrderMapper {
+    @Override
+    public OrderResponseDto mapToDto(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm");
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java]: let's make this a constant field

File: src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+import com.tickets.model.Order;
+import com.tickets.model.Ticket;
+import com.tickets.model.dto.OrderResponseDto;
+import com.tickets.service.mapper.OrderMapper;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java]:
```suggestion
@Component
```
this applies to all mappers

File: src/main/java/com/tickets/service/mapper/impl/UserMapperImpl.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package com.tickets.service.mapper.impl;
+
+import com.tickets.model.User;
+import com.tickets.model.dto.UserResponseDto;
+import com.tickets.service.mapper.UserMapper;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserMapperImpl implements UserMapper {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/impl/UserMapperImpl.java]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/tickets/controller/CinemaHallController.java [hunk 9]
```diff
@@ -27,15 +27,15 @@ public CinemaHallController(CinemaHallService cinemaHallService,
...
+        CinemaHall cinemaHall = cinemaHallMapper.mapToEntity(cinemaHallRequestDto);
         cinemaHallService.add(cinemaHall);
     }
 
     @GetMapping
     public List<CinemaHallResponseDto> getAll() {
         return cinemaHallService.getAll()
                 .stream()
-                .map(cinemaHallMapper::mapToDto)
+                .map(cinemaHallMapper::entityToMap)
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/controller/CinemaHallController.java]: what does `entityToMap` mean? are we creating a Map from Entity?
```suggestion
                .map(cinemaHallMapper::mapToDto)
```

File: src/main/java/com/tickets/controller/CinemaHallController.java [hunk 10]
```diff
@@ -27,15 +27,15 @@ public CinemaHallController(CinemaHallService cinemaHallService,
 
     @PostMapping
     public void addCinemaHall(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
-        CinemaHall cinemaHall = cinemaHallMapper.dtoToMap(cinemaHallRequestDto);
+        CinemaHall cinemaHall = cinemaHallMapper.mapToEntity(cinemaHallRequestDto);
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/controller/CinemaHallController.java:30]: this is ok - we are mapping dto to entity

File: src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+public class OrderMapperImpl implements OrderMapper {
+    private static final String DATE_FORMAT = "dd.MM.yyyy HH:mm";
+
+    @Override
+    public OrderResponseDto entityToMap(Order order) {
+        OrderResponseDto orderResponseDto = new OrderResponseDto();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
+        orderResponseDto.setOrderDate(order.getOrderDate().format(formatter));
+        orderResponseDto.setId(order.getId());
+        List<Long> ticketsId = order.getTickets()
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/impl/OrderMapperImpl.java]:
```suggestion
        List<Long> ticketIds = order.getTickets()
```

File: src/main/java/com/tickets/service/mapper/impl/ShoppingCartMapperImpl.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoppingCartMapperImpl implements ShoppingCartMapper {
+    @Override
+    public ShoppingCartResponseDto entityToMap(ShoppingCart shoppingCart) {
+        ShoppingCartResponseDto shoppingCartResponseDto = new ShoppingCartResponseDto();
+        shoppingCartResponseDto.setId(shoppingCart.getId());
+        List<Long> ticketsId = shoppingCart.getTickets()
```
Comment (KseniiaMakarova) [inline src/main/java/com/tickets/service/mapper/impl/ShoppingCartMapperImpl.java]:
```suggestion
        List<Long> ticketIds = shoppingCart.getTickets()
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
see a few comments above

**Pull request merged**